src/grt/src/GlobalRouter.cpp

```diff
@@ -3982,9 +3982,9 @@ void GlobalRouter::findBufferPinPostions(Net* net1,
                                          odb::Point& pin_pos2)
 {
   for (const Pin& pin1 : net1->getPins()) {
-    if (!pin1.isPort()) {
+    if (!pin1.isPort() && !pin1.isDeleted()) {
       for (const Pin& pin2 : net2->getPins()) {
-        if (!pin2.isPort()) {
+        if (!pin2.isPort() && !pin2.isDeleted()) {
           if (pin1.getITerm()->getInst() == pin2.getITerm()->getInst()) {
```

Comment:
@maliberty The problem resides here. When we delete an instance, the grt pin is not updated and tries to access an invalid pointer.

Maybe we can set the iterm_/bterm_ to nullptr when the pin is deleted and check it here. We have few places that calls the getITerm function and it would be easy to update them.